src/bank00.asm

```diff
@@ -9006,9 +9006,24 @@ getDialogTextInsertionPoint:
     ld   A, [wWindowTextSpaceLeftOnLine]               ;; 00:375f $fa $ba $d8
     ld   C, A                                          ;; 00:3762 $4f
     ret                                                ;; 00:3763 $c9
-    db   $d5, $3e, $7f, $15, $cd, $44, $38, $14        ;; 00:3764 ????????
-    db   $3e, $7f, $cd, $44, $38, $1c, $05, $20        ;; 00:376c ????????
-    db   $f0, $d1, $c9                                 ;; 00:3774 ???
+
+;copy and left shifts B bytes from HL in bank A to DE
+copyShiftBankAfromHLtoDE:
+    push HL                                            ;; 00:3764 $e5
+    call pushBankNrAndSwitch                           ;; 00:3765 $cd $fb $29
+    pop  HL                                            ;; 00:3768 $e1
+    ld   A, B                                          ;; 00:3769 $78
+    and  A, A                                          ;; 00:376a $a7
+    ret  Z                                             ;; 00:376b $c8
+.loop:
+    ld   A, [HL+]                                      ;; 00:376c $2a
+    add  A, A                                          ;; 00:376d $87
```

Comment:
It looks like this only works because the leftmost column of each number is white (the bit 7 of every byte is 0). Is that a conscious choice? I expect it is because there is no handling for overflowing to another tile.

If so, I think `rla` would make it clearer what is happening (even though I always have to look up shift/rotate instructions). Also a comment about the restriction at the top of the function.

Alternatively, as you know I've questioned whether this shift-by-one is necessary at all, so eliminating it is still an option.

Reply:
Yes. It's a shift instead of a roll. `sla` takes 2 cycles where `add A, A` only takes one. I could make it a roll if you want through `rlca`? (`rla` copies the carry bit into the LSB, which I don't think is wise for subsequent loop iterations).

It would have taken more code/functions to have separate left-aligned function logic for MP (especially since I fixed the symbol with the old drawNumberOnStatusBar function).

Reply:
I'll just make it a roll, that's probably slightly more useful than a shift there.